test/atomsbase.jl

```diff
@@ -0,0 +1,92 @@
+using Unitful
+using UnitfulAtomic
+
+@testset "AtomsBase support" begin
+    import AtomsBase
+    using AtomsBase: AbstractSystem, FlexibleSystem
+    using AtomsBaseTesting
+
+    function make_chemfiles_system(D=3; drop_atprop=Symbol[], infinite=false, kwargs...)
+        dropkeys = [:covalent_radius, :vdw_radius]  # Cannot be mutated in Chemfiles
+        data = make_test_system(D; drop_atprop=append!(drop_atprop, dropkeys),
+                                extra_sysprop=(; extra_data=42.0),
+                                cellmatrix=:upper_triangular,
+                                kwargs...)
+        if infinite
+            system = AtomsBase.isolated_system(data.atoms; data.sysprop...)
+        else
+            system = AtomsBase.periodic_system(data.atoms, data.box; data.sysprop...)
+        end
+        merge(data, (; system))
+    end
+
+    @testset "Conversion AtomsBase -> Chemfiles (periodic, velocity)" begin
+        system, atoms, atprop, sysprop, box, bcs = make_chemfiles_system(; infinite=false)
+        frame = convert(Frame, system)
+
+        D = 3
+        cell = Chemfiles.matrix(Chemfiles.UnitCell(frame))
+        for i in 1:D
+            @test cell[i, :] ≈ ustrip.(u"Å", box[i]) atol=1e-12
+        end
+        @test Chemfiles.shape(Chemfiles.UnitCell(frame)) in (Chemfiles.Triclinic,
+                                                             Chemfiles.Orthorhombic)
+        for (i, atom) in enumerate(frame)
+            @test(Chemfiles.positions(frame)[:, i]
+                  ≈ ustrip.(u"Å", atprop.position[i]), atol=1e-12)
```

Comment:
why an approximate equality? This should be exactly the same data, right?

Reply:
In theory yes, but this is an implementation detail (assumes that `FlexibleSystem`s don't internally do some conversion or sth.). So I'd rather have a bit of tolerance to avoid some random test failures in the future.

But your call: I can do exact comparison if you want.

Reply:
Just to be clear `FlexibleSystem` is a concrete implementation of the `AtomsBase` interface, which is provided by default by the `AtomsBase` package? In this case, I'd say we can keep this as is.

Reply:
Indeed :+1:, that's exactly as it is.